gui/static/js/components/scenarios.js

```diff
@@ -42,35 +44,39 @@ export default class ScenariosItem extends React.Component {
     }
 
     /* saves currently loaded scenario to file */
-    saveScenario(event){
+    saveScenario(){
         console.log("save scenario listener");
 
         var scenario = {items: this.state.items};
-        var filename = this.state.filename;
-
-        //Implemented the same way as Python script saving. This might not work so use axios code if it doesn't
-        event.preventDefault();
-        var element = document.createElement('a');
-        element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(JSON.stringify(this.state.items)));
-        element.setAttribute('download', this.state.filename);
-        element.style.display = 'none';
-        document.body.appendChild(element);
-        element.click();
-        document.body.removeChild(element);
-
-        // axios({
-        //     method:'POST',
-        //     url:'/saveScenario',
-        //     data: {scenario: scenario, name: filename},
-        //     dataType: 'text',
-        //     contentType: 'application/json; charset=utf-8'
-        // })
-        // .then(function(response) {
-        //     console.log('saved scenario successfully');
-        // })
-        // .catch(function (error) {
-        //     console.log(error);
-        // });
+        console.log("numBots:" + this.state.numBots);
+        if (this.state.numBots != 1) alert("There must be exactly one bot!");
+        else {
+            var filename = this.state.filename;
+
+            //Implemented the same way as Python script saving. This might not work so use axios code if it doesn't
+            //event . preventDefault ()
```

Comment:
Please remove if it is not necessary.

Reply:
We don't know if it's necessary or not until we test it, and it's not ready to test just yet.

Reply:
deleted; I realized that I assigned filename a value in the constructor